File: src/runtime/boxes/fail/bb_fail.java
package bb;

/**
 * bb_fail.java — FAIL: always ω — force backtrack
 * Port of bb_fail.c / bb_fail.s
 */
public class bb_fail extends bb_box {

    public bb_fail(MatchState ms) { super(ms); }

    @Override public Spec α() { return null; }
    @Override public Spec β()  { return null; }
}

File: src/runtime/boxes/fail/bb_fail.cs
// bb_fail.cs — FAIL: always ω — force backtrack
// Mirrors src/runtime/boxes/fail/bb_fail.c

namespace Snobol4.Runtime.Boxes;

public sealed class bb_fail : IByrdBox
{
    public Spec Alpha(MatchState ms) => Spec.Fail;
    public Spec Beta(MatchState ms)  => Spec.Fail;
}
